src/response.rs

```diff
@@ -32,13 +31,54 @@ where
     raw_body_read: usize,
 }
 
+#[cfg(feature = "defmt")]
+impl<C> defmt::Format for Response<'_, '_, C>
+where
+    C: Read + Write,
+{
+    fn format(&self, fmt: defmt::Formatter) {
+        defmt::write!(
+            fmt,
+            "Response {{ method = {}, status = {}, content_type = {}, content_length = {}, transfer_encoding = {}, keep_alive = {}, header_buf = {:?}, header_len = {}, raw_body_read = {} }}",
+            self.method,
+            self.status,
+            self.content_type,
+            self.content_length,
+            self.transfer_encoding,
+            self.keep_alive,
+            self.header_buf,
+            self.header_len,
+            self.raw_body_read,
+        )
+    }
+}
+
+impl<C> core::fmt::Debug for Response<'_, '_, C>
+where
+    C: Read + Write,
+{
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("Response")
+            .field("method", &self.method)
+            .field("status", &self.status)
+            .field("content_type", &self.content_type)
+            .field("content_length", &self.content_length)
+            .field("transfer_encoding", &self.transfer_encoding)
+            .field("keep_alive", &self.keep_alive)
+            .field("header_buf", &self.header_buf)
+            .field("header_len", &self.header_len)
+            .field("raw_body_read", &self.raw_body_read)
+            .finish()
+    }
+}
+
 impl<'buf, 'conn, C> Response<'buf, 'conn, C>
 where
-    C: Read,
+    C: Read + Write,
```

Comment:
Is there any reason why the connection on the response now must be `Write`?

Reply:
Yes, unfortunately the `C` in `HttpConnection<'conn, C>` is `Read + Write` and I think it goes up to `TcpConnect::Connection` being `Read+Write`.

Reply:
Heh, I guess there isn't any reason. Good leading question, it took me way too long to figure out, but whatever. The result is even simpler than before the PR.